contracts/Zauction.sol

```diff
@@ -35,30 +37,44 @@ contract ZAuction is Initializable, OwnableUpgradeable {
     uint256 amount,
     address nftAddress,
     uint256 tokenId,
-    uint256 expireBlock
+    uint256 expireBlock,
+    address paymentToken,
+    uint256 topLevelDomainId
   );
 
   event DomainSold(
     address indexed buyer,
     address indexed seller,
     uint256 amount,
     address nftAddress,
-    uint256 indexed tokenId
+    uint256 indexed tokenId,
+    address paymentToken,
+    uint256 topLevelDomainId
   );
 
-  event BuyNowPriceSet(uint256 indexed tokenId, uint256 amount);
+  event BuyNowPriceSet(
+    uint256 indexed tokenId,
+    uint256 amount,
+    address paymentToken
+  );
 
   event BidCancelled(uint256 bidNonce, address indexed bidder);
 
   // On upgrade new variables must be inserted after all existing variables
   IZNSHub public hub;
 
+  // Map top level domain ID to the ERC20 token used for that network.
+  mapping(uint256 => IERC20) public networkToken;
+
+  // Hardcoded address of the wild token for legacy bids only
+  IERC20 private wildToken;
+
   function initialize(IERC20 tokenAddress, IRegistrar registrarAddress)
     public
     initializer
   {
     __Ownable_init();
-    token = tokenAddress;
+    defaultToken = tokenAddress;
     registrar = registrarAddress;
   }
```

Comment:
Create a `upgradeFromV2` function that allows for `defaultToken` and `wildToken` to be set.
This function should only be callable by the address found at `_getAdmin()` via:

```
// Open Zeppelin StorageSlot Library
import {StorageSlot} from "../oz/utils/StorageSlot.sol";

/**
   * @dev Storage slot with the proxy admin of the contract.
   */
bytes32 internal constant _ADMIN_SLOT =
  0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103;

function _getAdmin() internal view returns (address) {
  return StorageSlot.getAddressSlot(_ADMIN_SLOT).value;
}

```